api.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import logging
+import tornado
+import signal
+
+from logging.handlers import TimedRotatingFileHandler
+from pymongo import MongoClient
+from queue import Queue
+
+import config
+os.makedirs(config.LOGS_DIR, exist_ok=True)
+logging.basicConfig(
+    handlers=[
+        TimedRotatingFileHandler(
+            "{}/log".format(config.LOGS_DIR), 
+            when=config.LOGS_ROTATE_WHEN, 
+            backupCount=config.LOGS_BACKUP_COUNT
+        ),
+        logging.StreamHandler()
+    ],
+    level=logging.INFO if not config.DEBUG else logging.DEBUG
+)
+
+from broadway_api.callbacks import *
+from broadway_api.handlers.client import *
+from broadway_api.handlers.worker import *
+from broadway_api.utils.bootstrap import *
+
+logger = logging.getLogger("api")
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--course-config", help="A JSON file that specifies the courses and their tokens"
+    )
+    parser.add_argument(
+        "--https", action="store_true", help="Use HTTPS to serve requests"
+    )
+    parser.add_argument("--ssl-certificate", help="Path to the SSL certificate")
+    parser.add_argument("--ssl-key", help="Path to the SSL private key file")
+    return parser.parse_args()
+
+def initialize_app():
+    config_keys = filter(lambda x: not x.startswith("_"), config.__dict__)
```

Comment:
We should probably document this somewhere to avoid the oddity that naming a config variable `_SOMETHING` causes a soft error.